plugins/catalog-backend-module-gitlab/src/GitLabDiscoveryProcessor.ts

```diff
@@ -97,14 +98,25 @@ export class GitLabDiscoveryProcessor implements CatalogProcessor {
       config: integration.config,
       logger: this.logger,
     });
-    const startTimestamp = Date.now();
     this.logger.debug(`Reading GitLab projects from ${location.target}`);
 
-    const projects = paginated(options => client.listProjects(options), {
+    const opts = {
       group,
-      last_activity_after: await this.updateLastActivity(),
       page: 1,
-    });
+    } as {
+      group: string;
+      page: number;
+      last_activity_after: string | undefined;
+    };
+
+    const lastActivity = (await this.cache.get(this.getCacheKey())) as string;
+    // We check for the existence of lastActivity and only set it if it's present to ensure
+    // that the options doesn't include the key so that the API doesn't receive an empty query parameter.
+    if (lastActivity) {
+      opts.last_activity_after = lastActivity;
+    }
```

Comment:
Think it should be possible to do something like this instead:

```ts
const lastActivity = await this.cache.get(this.getCacheKey());

const opts = {
  group,
  page: 1,
  ...(lastActivity && { last_activity_after: lastActivity }),
}
```

Reply:
Fixed this.